Sources/OpenTelemetry/Tracing/Processing/OTelMultiplexSpanProcessor.swift

```diff
@@ -0,0 +1,74 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift OTel open source project
+//
+// Copyright (c) 2023 Moritz Lang and the Swift OTel project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import ServiceContextModule
+import ServiceLifecycle
+
+/// A pseudo-``OTelSpanProcessor`` that may be used to process using multiple other ``OTelSpanProcessor``s.
+public actor OTelMultiplexSpanProcessor: OTelSpanProcessor {
+    private let processors: [any OTelSpanProcessor]
+    private let shutdownStream: AsyncStream<Void>
+    private let shutdownContinuation: AsyncStream<Void>.Continuation
+
+    /// Create an ``OTelMultiplexSpanProcessor``.
+    ///
+    /// - Parameter processors: An array of ``OTelSpanProcessor``s, each of which will be invoked on start and end of spans.
+    /// Processors are called sequentially and the order of this array defines the order in which they're being called.
+    public init(processors: [any OTelSpanProcessor]) {
+        self.processors = processors
+        (shutdownStream, shutdownContinuation) = AsyncStream.makeStream()
+    }
+
+    public func run() async throws {
+        await withThrowingTaskGroup(of: Void.self) { group in
+            group.addTask {
+                var shutdowns = self.shutdownStream.makeAsyncIterator()
+                await shutdowns.next()
```

Comment:
I guess this comes back to one of my previous questions. Why do the processors have a `shutdown` method instead of a `run` + ServiceLifecycle?

Reply:
Do you mean the exporters? I was thinking about that too, but they don't necessarily have a run-loop so I thought it wouldn't be needed. [`OTLPGRPCSpanExporter`](https://github.com/slashmo/swift-otel/blob/next/Sources/OTLPGRPC/Tracing/OTLPGRPCSpanExporter.swift), e.g., uses a generated gRPC client to send span batches. I wouldn't know what to put in its `run` method apart from the `AsyncStream` dance I did for the no-ops above. Plus, I don't expect people to put an exporter into a `ServiceGroup` anyways since they're always used inside span processors.